extensions/machine-learning-services/src/test/dialogs/externalLanguages/languageEditDialog.test.ts

```diff
@@ -0,0 +1,50 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as should from 'should';
+import 'mocha';
+import { createContext, ParentDialog } from './utils';
+import { LanguageEditDialog } from '../../../dialogs/externalLanguages/languageEditDialog';
+import { LanguageUpdateModel } from '../../../dialogs/externalLanguages/languageViewBase';
+
+describe('Edit External Languages Dialog', () => {
+	it('Should open dialog successfully ', async function (): Promise<void> {
+		let testContext = createContext();
+		let parent = new ParentDialog(testContext.apiWrapper.object);
+		let languageUpdateModel: LanguageUpdateModel = {
+			content: parent.createNewContent(),
+			language: parent.createNewLanguage(),
+			newLang: true
+		};
+		let dialog = new LanguageEditDialog(testContext.apiWrapper.object, parent, languageUpdateModel);
+		dialog.showDialog();
+		should.notEqual(dialog.addNewLanguageTab, undefined);
+	});
+
+	it('Should raise save event if save button clicked ', async function (): Promise<void> {
+		let testContext = createContext();
+		let parent = new ParentDialog(testContext.apiWrapper.object);
+		let languageUpdateModel: LanguageUpdateModel = {
+			content: parent.createNewContent(),
+			language: parent.createNewLanguage(),
+			newLang: true
+		};
+		let dialog = new LanguageEditDialog(testContext.apiWrapper.object, parent, languageUpdateModel);
+		dialog.showDialog();
+
+		let updateCalled = false;
+		let promise = new Promise(resolve => {
```

Comment:
This shouldn't really be necessary - if update isn't called this will never resolve and so the test will time out first anyways

Reply:
it's an even handler so if I don't do this can make the test flaky 

---
In reply to: [375369484](https://github.com/microsoft/azuredatastudio/pull/9043#discussion_r375369484) [](ancestors = 375369484)